### PR TITLE
Merge required branches before Substrate updates

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

File: pallets/permissions/src/default_permissions.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+use crate::{SpacePermission as SP, SpacePermissions, SpacePermissionSet};
+
+#[allow(unused_imports)]
```
Comment (F3Joule) [inline pallets/permissions/src/default_permissions.rs:3]:
```suggestion
```

File: pallets/permissions/src/default_permissions.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      SP::HideAnyPost,
+      SP::HideAnyComment,
+
+      SP::SuggestEntityStatus,
+      SP::UpdateEntityStatus,
+
+      SP::UpdateSpaceSettings,
+    ].into_iter())),
+  };
+}
```
Comment (F3Joule) [inline pallets/permissions/src/default_permissions.rs:64]:
```suggestion
}

```

File: pallets/posts/src/functions.rs [hunk 3]
```diff
@@ -318,6 +318,77 @@ impl<T: Trait> Module<T> {
...
+        let new_space = Spaces::<T>::require_space(new_space_id)?;
+
+        Spaces::ensure_account_has_space_permission(
+            editor,
+            &new_space,
+            SpacePermission::CreatePosts,
+            Error::<T>::NoPermissionToCreatePosts.into()
+        )?;
+
+        // TODO check whether post and its content are not blocked within a new space
```
Comment (F3Joule) [inline pallets/posts/src/functions.rs:351]: There should be a way to set up checks here without implementing pallet-moderation into the Runtime

File: pallets/session-keys/Cargo.toml [hunk 4]
```diff
@@ -61,12 +64,64 @@ git = 'https://github.com/paritytech/substrate.git'
...
+
 [dependencies.pallet-utils]
 default-features = false
 path = '../utils'
-version = '0.5.0'
+
+[dependencies.pallet-spaces]
+default-features = false
+path = '../spaces'
+version = '0.4.0'
```
Comment (F3Joule) [inline pallets/session-keys/Cargo.toml:81]:
```suggestion
```

File: pallets/session-keys/Cargo.toml [hunk 5]
```diff
@@ -61,12 +64,64 @@ git = 'https://github.com/paritytech/substrate.git'
...
+
+[dependencies.pallet-space-follows]
+default-features = false
+path = '../space-follows'
+version = '0.4.0'
```
Comment (F3Joule) [inline pallets/session-keys/Cargo.toml:86]:
```suggestion
```

File: pallets/session-keys/Cargo.toml [hunk 6]
```diff
@@ -61,12 +64,64 @@ git = 'https://github.com/paritytech/substrate.git'
...
+version = '2.0.0-rc4'
+
+[dev-dependencies.pallet-roles]
+default-features = false
+path = "../roles"
+
+[dev-dependencies.df-traits]
+default-features = false
+path = '../traits'
+version = '0.4.0'
```
Comment (F3Joule) [inline pallets/session-keys/Cargo.toml:117]:
```suggestion
```

File: runtime/Cargo.toml [hunk 7]
```diff
@@ -44,7 +44,7 @@ std = [
     'pallet-space-follows/std',
     'pallet-space-ownership/std',
     'pallet-space-history/std',
-#    'pallet-session-keys/std',
+#    'session-keys/std',
```
Comment (F3Joule) [inline runtime/Cargo.toml:47]:
```suggestion
#    'pallet-session-keys/std',
```

**Comment by siman** [comment]
4 failures in df-integration-tests:

```
failures:

---- tests::move_hidden_post_should_fail_cannot_move_comment stdout ----
thread 'tests::move_hidden_post_should_fail_cannot_move_comment' panicked at 'assertion failed: `(left == right)`
  left: `Err(DispatchError::Module { index: 0, error: 0, message: Some("AccountIsBlocked") })`,
 right: `Err(DispatchError::Module { index: 0, error: 10, message: Some("CannotUpdateSpaceIdOnComment") })`', pallets/df-integration-tests/src/lib.rs:2010:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::move_hidden_post_should_fail_no_permission_to_create_post stdout ----
thread 'tests::move_hidden_post_should_fail_no_permission_to_create_post' panicked at 'assertion failed: `(left == right)`
  left: `Err(DispatchError::Module { index: 0, error: 0, message: Some("AccountIsBlocked") })`,
 right: `Err(DispatchError::Module { index: 0, error: 14, message: Some("NoPermissionToCreatePosts") })`', pallets/df-integration-tests/src/lib.rs:1998:13

---- tests::move_post_should_work stdout ----
thread 'tests::move_post_should_work' panicked at 'Expected Ok(_). Got Err(
    DispatchError::Module {
        index: 0,
        error: 0,
        message: Some(
            "AccountIsBlocked",
        ),
    },
)', pallets/df-integration-tests/src/lib.rs:1919:13

---- tests::move_hidden_post_should_work stdout ----
thread 'tests::move_hidden_post_should_work' panicked at 'Expected Ok(_). Got Err(
    DispatchError::Module {
        index: 0,
        error: 0,
        message: Some(
            "AccountIsBlocked",
        ),
    },
)', pallets/df-integration-tests/src/lib.rs:1955:13


failures:
    tests::move_hidden_post_should_fail_cannot_move_comment
    tests::move_hidden_post_should_fail_no_permission_to_create_post
    tests::move_hidden_post_should_work
    tests::move_post_should_work

test result: FAILED. 136 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by F3Joule** [comment]
> 4 failures in df-integration-tests:

Fixed already

**Review by subsocialdev — CHANGES_REQUESTED** [review]

**Pull request merged**